mesheryctl/internal/cli/root/system/check.go

```diff
@@ -39,19 +39,33 @@ var PreCheckCmd = &cobra.Command{
 
 //Run preflight healthchecks to verify environment health
 func runPreflightHealthChecks(isPreRunExecution bool, subcommand string, platform string) error {
-	//Run docker healthchecks
-	if err := runDockerHealthCheck(isPreRunExecution, subcommand, platform); err != nil {
-		return err
+	//Docker healthchecks are only invoked when it's not a PreRunExecution
+	// or it's a PreRunExecution and current platform is docker
+	if !isPreRunExecution || (isPreRunExecution && platform == "docker") {
+		//Run docker healthchecks
+		if err := runDockerHealthCheck(isPreRunExecution, subcommand, platform); err != nil {
+			return err
+		}
 	}
-	//Run k8s API healthchecks
-	if err = runKubernetesAPIHealthCheck(); err != nil {
-		return err
+	//Kubernetes healthchecks are only invoked when it's not a PreRunExecution
+	// or it's a PreRunExecution and current platform is kubernetes
+	if !isPreRunExecution || (isPreRunExecution && platform == "kubernetes") {
+		//Run k8s API healthchecks
+		if err = runKubernetesAPIHealthCheck(isPreRunExecution); err != nil {
+			return err
+		}
 	}
-	//Run k8s plus kubectl minimum version healthchecks
-	if err := runKubernetesVersionHealthCheck(); err != nil {
-		return err
+	//KubernetesVersion healthchecks are only invoked when it's not a PreRunExecution
+	// or it's a PreRunExecution and current platform is kubernetes
+	if !isPreRunExecution || (isPreRunExecution && platform == "kubernetes") {
+		//Run k8s plus kubectl minimum version healthchecks
+		if err := runKubernetesVersionHealthCheck(isPreRunExecution); err != nil {
```

Comment:
@piyushsingariya , Move this under the above if block ? It would also show the continuity of checks done for k8s platform.

Reply:
Nice observation! I'll change it.